src/filter/CategoricalFilter.ts

```diff
@@ -83,26 +85,19 @@ export default class CategoricalFilter extends AVectorFilter<string, ICategorica
   private async generateCategories($node: d3.Selection<any>, dispHistogram: boolean) {
     const that = this;
     const cellHeight = this.filterDim.height;
-    const allCatNames = await(<any>this.data).data();
-    const categories = (<any>this.data).desc.value.categories;
-
-    const c20 = d3.scale.category20();
-    const toolTip = (this.generateTooltip($node));
-    const cellDimension = this.filterDim.width / categories.length;
+    const hist = <ICatHistogram> await this.data.hist();
+    const bins = [];
+    hist.forEach((d, i) => bins.push(d));
     const catData = [];
-    const uniqueCategories = allCatNames.filter((x, i, a) => a.indexOf(x) === i);
-    uniqueCategories.forEach(((val, i) => {
-      const count = allCatNames.filter(isSame.bind(this, val));
-      let colcat = [];
-      if (typeof categories !== 'undefined') {
-        colcat = categories.filter((d, i) => {
-          return d.name === val;
-        });
-      }
-      const colorVal = (colcat.length < 1) ? c20(count.length) : colcat[0].color;
-      catData.push({name: val, count: count.length, 'color': colorVal});
-    }));
-
+    const categories = (<any>this.data.desc.value).categories;
```

Comment:
why is this any cast needed? shouldn't this.data already be a `ICategoricalVector`?